### PR TITLE
HTML parser initial version

### PR DESCRIPTION
**Pull request by BalazsAtWork**

### CODE REVIEW

File: livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        val tableElements = document.getElementsByTag("table")
+        return parseTableElements(tableElements)
+    }
+
+    private fun parseTableElements(tableElements: Elements): List<DocumentNode> {
+        fun tableHasAtLeastTwoRows(table: Element) = table.getElementsByTag("tr").size > 1
+
+        return tableElements
+                .filter(::tableHasAtLeastTwoRows)
+                .map { parseTableToDecisionTable(it) }
```
Comment (Alexander-Miller) [inline livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt]: Why the inconsistent style?

File: livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        val tableRows = table.getElementsByTag("tr")
+        val headers = extractHeadersFromFirstRow(tableRows)
+        val dataRows = parseDataRow(headers, tableRows)
+        return DocumentNode.DecisionTable(headers, dataRows, emptyList())
+    }
+
+    private fun extractHeadersFromFirstRow(tableRows: Elements): List<String> {
+        val firstRowContainingHeaders = tableRows[0]
+        val headers = firstRowContainingHeaders.children()
+                .filter(::isHeaderOrDataCell)
```
Comment (Alexander-Miller) [inline livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt]: Same.

File: livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                .map { it.text() }.toList()
+
+        if (headers.size != headers.distinct().size) {
+            throw ParseException("Headers must contains only unique values: " + headers)
+        }
+        return headers
+    }
+
+    private fun parseDataRow(headers: List<String>, tableRows: Elements): List<DocumentNode.DecisionTableRow> {
+        val dataRows = ArrayList<DocumentNode.DecisionTableRow>()
```
Comment (Alexander-Miller) [inline livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt]: mutableListOf is more idiomatic.

File: livingdoc-repository-file/src/test/kotlin/org/livingdoc/repositories/format/HtmlFormatTest.kt [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+    @Test fun `exception if headers are not unique`() {
+        var exception = assertThrows(ParseException::class.java) {
+            cut.parse(getHtmlTableWithNonUniqueHeaders().byteInputStream())
+        }
+
+        assertThat(exception).hasMessageStartingWith("Headers must contains only unique values:")
+    }
+
+    @Test fun `exception if header count not equals to data cell count`() {
```
Comment (Alexander-Miller) [inline livingdoc-repository-file/src/test/kotlin/org/livingdoc/repositories/format/HtmlFormatTest.kt]: equal

File: livingdoc-repositories/src/main/kotlin/org/livingdoc/repositories/Document.kt [hunk 5]
```diff
@@ -11,18 +11,19 @@ sealed class DocumentNode {
...
+    data class DecisionTable(
+            val headers: List<String>,
+            val rows: List<DecisionTableRow>,
             override val children: List<DocumentNode>
     ) : DocumentNode()
 
-    data class TableRow(
-            val cells: List<TableCell>,
+    data class DecisionTableRow(
+            val cells: Map<String, TableCell>,
```
Comment (Alexander-Miller) [inline livingdoc-repositories/src/main/kotlin/org/livingdoc/repositories/Document.kt]: Should also rename it to DecisionTableCell.

File: livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt (hunk 1 above)
Comment (BalazsAtWork) [reply livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt]: IntelliJ formatter

File: livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt (hunk 2 above)
Comment (BalazsAtWork) [reply livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt]: Same

File: livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt (hunk 3 above)
Comment (BalazsAtWork) [reply livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt]: OK

File: livingdoc-repository-file/src/test/kotlin/org/livingdoc/repositories/format/HtmlFormatTest.kt (hunk 4 above)
Comment (BalazsAtWork) [reply livingdoc-repository-file/src/test/kotlin/org/livingdoc/repositories/format/HtmlFormatTest.kt]: OK

File: livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt (hunk 2 above)
Comment (BalazsAtWork) [reply livingdoc-repository-file/src/main/kotlin/org/livingdoc/repositories/format/HtmlFormat.kt]: DONE

**Pull request merged**